/* references
    - guide: https://developers.google.com/protocol-buffers/docs/proto
    - using timestamp: https://stackoverflow.com/questions/3574716/date-and-time-type-for-use-with-protobuf

*/

syntax = "proto3";
package sensors;

message Gps{
    
    double latitude = 1;
    double longitude = 2;
    double horizontalDilutionOfPrecision = 3;

    double timestamp=4;
}

message Imu{
    float scaledAccelX = 1;
    float scaledAccelY = 2;
    float scaledAccelZ = 3;

    double timestamp=4;
}

message Cam{
    int32 cols = 1;
    int32 rows = 2;
    bytes image_data = 3;

    double timestamp=4;
}

message Lidar{
    float size=1;
     message xyz{
    float x=1;
    float y=2;
    float z=3;
    }
    repeated xyz data=2;

    double timestamp=3;
}

//enum{HANDLE_ANGLE, TURN_LIGHT, RPM, GEAR};
message Can{
   float type=1;
   float data=2;

   double timestamp=3;
}